src/DonorProfiles/Repositories/Donations.php

```diff
@@ -21,10 +24,11 @@ public function getDonationCount( $donorId ) {
 				"
 				SELECT count(revenue.id) as count
 				FROM {$wpdb->give_revenue} as revenue
-				INNER JOIN {$wpdb->posts} as posts
-				ON revenue.donation_id = posts.ID
-				WHERE posts.post_author = %d
-				AND posts.post_status IN ( 'publish', 'give_subscription' )
+				INNER JOIN {$wpdb->posts} as posts ON revenue.donation_id = posts.ID
+				INNER JOIN {$wpdb->prefix}give_donationmeta as donationmeta ON revenue.donation_id = donationmeta.donation_id
+				WHERE donationmeta.meta_key = '_give_payment_donor_id'
+					AND donationmeta.meta_value = %d
+					AND posts.post_status IN ( 'publish', 'give_subscription' )
```

Comment:
All of these queries are identical aside from the SELECT. I suggest making a private `Donations::getDonationAggregate` method:
```php
private function getDonationAggregate($rawAggregate) {
	return DB::get_row(
		DB::prepare("
			SELECT {$rawAggregate}
			FROM {$wpdb->give_revenue} as revenue
				INNER JOIN {$wpdb->posts} as posts ON revenue.donation_id = posts.ID
				INNER JOIN {$wpdb->prefix}give_donationmeta as donationmeta ON revenue.donation_id = donationmeta.donation_id
			WHERE donationmeta.meta_key = '_give_payment_donor_id'
				AND donationmeta.meta_value = %d
				AND posts.post_status IN ( 'publish', 'give_subscription' )
		");
	);
}
```

Used as:
```php
return $this->getDonationAggregate('count(revenue.id)');
```

Reply:
Good thinking, and glad to get the DB class into play here. Done, ref: https://github.com/impress-org/givewp/pull/5569/commits/b00923a6246af75dd9c941d5d3c12fd622012245